.github/workflows/documentation.yaml

```diff
@@ -14,10 +14,10 @@ jobs:
 
     steps:
     - uses: actions/checkout@v3
-    - name: Set up Python 3.8
```

Comment:
Please segregate the changes to GitHub actions into a separate MR. These changes are out of scope of this MR.

Reply:
🤔 But I did change the python version to 3.10 in `pyproject.toml` because the `kwonly` feature of dataclass is only available to versions after 3.10. Does this matter? (Without `kwonly=True` the `OpenSourceConfig` cannot have non-default values like `model_path`)

Reply:
Definitely using a 3.10 feature of kwonly cannot be the reason to so easily bump the python version for the entire project. :D Can you rework the code to be 3.8 compatible? I am fine if you make a follow up discussion about why you need kwonly so badly that you want a bump to 3.10, then we will discuss it.